src/JustWatch.elm

```diff
@@ -0,0 +1,223 @@
+module JustWatch exposing (..)
+
+import Json.Encode as Encode
+import Json.Decode as Decode exposing (string, list)
+import Json.Decode.Pipeline exposing (decode, hardcoded, optional, required)
+import Http exposing (encodeUri)
+
+
+type MovieOffers
+    = Loading
+    | NoResults
+    | Results (List Offer)
+
+
+type OfferType
+    = Buy
+    | Rent
+    | Streaming
+    | Ads
+    | Free
+    | Unknown
+
+
+movieSearchUrl : String -> String
+movieSearchUrl title =
+    -- generic JSON object
+    Encode.object [ ( "query", Encode.string title ) ]
+        -- Encode to json { "query": "thetitle" }
+        |> Encode.encode 0
+        -- Encode for uri
+        |> Http.encodeUri
+        |> String.append "https://apis.justwatch.com/content/titles/en_US/popular?body="
+
+
+movieDetailUrl : Int -> String
+movieDetailUrl id =
+    "https://apis.justwatch.com/content/titles/movie/" ++ (toString id) ++ "/locale/en_US"
+
+
+type alias SearchResult =
+    { title : String
+    , id : Int
+    }
+
+
+
+--
+-- decodeConditions : Decoder (List ConditionResult)
+-- decodeConditions=
+--     decode ConditionResult
+--         |> required "id" int
+--         |> Json.Decode.list
+
+
+decodeSearch : Decode.Decoder (List SearchResult)
+decodeSearch =
+    Decode.at [ "items" ] (Decode.list searchResultDecoder)
```

Comment:
This too me way to long to figure out.

Reply:
My experience with decoders was like that until a few different things 'clicked' and they've been pretty easy since then (so far)